packages/devextreme/js/data/query.d.ts

```diff
@@ -166,13 +167,26 @@ export interface Query {
     toArray(): Array<any>;
 }
 
+/**
+ * @docid
+ * @type object
+ * @public
+ */
+export interface QueryOptions {
+    /**
+     * @docid
+     * @public
+     */
+    langParams?: LangParams;
```

Comment:
Question.

We adding only the ```case-sensitive/case-insensitive``` comparison feature to the ```DataQuery```
But here we declare a wide ```LangParams``` type.

The ```LangParams.locale``` looks ok because this options is taken into account in comparison [here](https://github.com/DevExpress/DevExtreme/blob/24_2/packages/devextreme/js/core/utils/data.js#L200)
But the ```LangParams.collatorOptions``` has a lot of parameters that the ```DataQuery``` doesn't take into account as far as I can see (I may be wrong)

E.g. the ```sensitivity``` has four different variants ([link to the docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Collator/Collator#sensitivity))
The ```DataQuery``` takes into account only the ```case-sensitive/case-insensitive```

So, maybe it's a good idea to make this type more strict and custom?
Something like this:
```typescript
langOptions?: {
  locale: string; // like in LangParams
  caseSensitive: boolean; 
}
```

@tomodasheesh, @GoodDayForSurf, @VasilyStrelyaev, what do you think about it? :)

Reply:
why we add type for query? query can be ArrayQuery or remoteQuery. QueryOptions is not used in remoteQuery..

Reply:
We declare a wide `LangParams` type since this utility function requires `langParams` as the third argument
https://github.com/DevExpress/DevExtreme/blob/b9a7f67445d640c32780acfbff39453c6f4a8c1f/packages/devextreme/js/data/array_query.js#L245

I just saw that we also use queryOptions.errorHandler, so we should take that into account as well.
https://github.com/DevExpress/DevExtreme/blob/b9a7f67445d640c32780acfbff39453c6f4a8c1f/packages/devextreme/js/data/array_query.js#L552

Regarding remoteQuery and arrayQuery, maybe I'll change the type name to be specific to array_query only?